### ISSUE TITLE
Sent 1000 more request in parallel , get 500 internal error rather than get 429 or 503 code

### ISSUE THREAD

**Issue by shinjikao** [OneDrive/onedrive-api-docs#760]
https://docs.microsoft.com/zh-tw/onedrive/developer/rest-api/concepts/errors
Following above docs, I want write a unit-test for cause a "too many requests" error in OneDrive, like below:
```

@Test
    public void test() throws Exception {
        int threadCount = 1000;
        Callable<Document> task = new Callable<Document>() {
            @Override
            public Document call() throws Exception {
                Timber.e("call");
                oneDriveProvider.getDocument("9005F332ACBDB318%2142511");
                oneDriveProvider.getDocument("9005F332ACBDB318%2142511");
                return oneDriveProvider.getDocument("9005F332ACBDB318%2142513");
            }
        };

        List<Callable<Document>> tasks = Collections.nCopies(threadCount, task);
        ExecutorService executorService = Executors.newFixedThreadPool(threadCount);
        List<Future<Document>> futures = executorService.invokeAll(tasks);
        List<Document> resultList = new ArrayList<>(futures.size());
        for (Future<Document> future : futures) {
            resultList.add(future.get());
        }
        assertNotNull(resultList);
    }
```

I always get code is 500, message is Internal Server Error,body is {"error":{"code":"generalException","message":"An error occurred in the data store."}}, but I expected I can get 429 or 503 and including "retry-after" header in body, who can investigate this?

**Comment by shinjikao** [comment]
I current use endpoint is 
https://api.onedrive.com/v1.0/

**Comment by ificator** [comment]
Hi @shinjikao, I assume the requests ordinarily succeed and it's only after some number of requests that you eventually get the 500s in this scenario?

**Comment by shinjikao** [comment]
Hi @ificator , I expected get 429 or 503 and including "retry-after" when number of requests so that I can handle it by "retry-after", but almost get 500 error now.

**Comment by shinjikao** [comment]
@ificator any update?

**Comment by patrick-rodgers** [comment]
As part of a repository clean up effort we are closing older issues. If this issue remains, please: open a new issue, reference this issue, and provide any additional details that may help in resolution. Thank you for your understanding as we work to improve our responsiveness.
